src/black/nodes.py

```diff
@@ -932,9 +932,14 @@ def is_type_comment(leaf: Leaf) -> bool:
     be used for general type comments (excluding ignore annotations, which should
     use `is_type_ignore_comment`). Note that general type comments are no longer
     used in modern version of Python, this function may be deprecated in the future."""
-    t = leaf.type
-    v = leaf.value
-    return t in {token.COMMENT, STANDALONE_COMMENT} and v.startswith("# type:")
+    if leaf.type not in {token.COMMENT, STANDALONE_COMMENT}:
+        return False
+    comment_text = leaf.value.lstrip("#").lstrip()
+    if not comment_text.startswith("type:"):
+        return False
+    type_annotation = comment_text[5:].strip()
+    leaf.value = f"# type: {type_annotation}" if type_annotation else "# type:"
```

Comment:
I think this function should probably stay pure, maybe put the modification part somewhere else

Also, this will need to go in the preview style